File: Examples/CSharp/SpotAccountTrade/CancelAnExistingOrderAndSendANewOrder_Example.cs
namespace Binance.Spot.SpotAccountTradeExamples
{
    using System;
    using System.Net;
    using System.Net.Http;
    using System.Threading.Tasks;
    using Binance.Common;
    using Binance.Spot;
    using Binance.Spot.Models;
    using Microsoft.Extensions.Logging;

    public class CancelAnExistingOrderAndSendANewOrder_Example
    {
        public static async Task Main(string[] args)
        {
            using var loggerFactory = LoggerFactory.Create(builder =>
            {
                builder.AddConsole();
            });
            ILogger logger = loggerFactory.CreateLogger<CancelAnExistingOrderAndSendANewOrder_Example>();

            HttpMessageHandler loggingHandler = new BinanceLoggingHandler(logger: logger);
            HttpClient httpClient = new HttpClient(handler: loggingHandler);

            string apiKey = "api-key";
            string apiSecret = "api-secret";

            var spotAccountTrade = new SpotAccountTrade(httpClient, apiKey, apiSecret);

            var result = await spotAccountTrade.CancelAnExistingOrderAndSendANewOrder("BNBUSDT", Side.SELL, OrderType.LIMIT, "STOP_ON_FAILURE", cancelOrderId: 12, timeInForce: TimeInForce.GTC, quantity: 10.1m, price: 295.92m);
        }
    }
}

File: Examples/FSharp/SpotAccountTrade/CancelAnExistingOrderAndSendANewOrder_Example.fs
open System
open System.Net
open System.Net.Http
open System.Threading.Tasks
open Microsoft.Extensions.Logging
open Binance.Common
open Binance.Spot
open Binance.Spot.Models

[<EntryPoint>]
let main argv =
    let loggerFactory = LoggerFactory.Create(fun (builder:ILoggingBuilder) -> 
        builder.AddConsole() |> ignore
    )
    let logger = loggerFactory.CreateLogger()

    let loggingHandler = new BinanceLoggingHandler(logger)
    let httpClient = new HttpClient(loggingHandler)

    let apiKey = "api-key";
    let apiSecret = "api-secret";
    
    let spotAccountTrade = new SpotAccountTrade(httpClient, apiKey = apiKey, apiSecret = apiSecret)
    
    let result = spotAccountTrade.CancelAnExistingOrderAndSendANewOrder("BNBUSDT", Side.SELL, OrderType.LIMIT, "STOP_ON_FAILURE", cancelOrderId = 12, timeInForce = TimeInForce.GTC, quantity = 10.1m, price = 295.92m) |> Async.AwaitTask |> Async.RunSynchronously
    
    0
